### PR TITLE
Feature/toast

### PR DESCRIPTION
**Pull request by brandonjmckay**
Continuation of https://github.com/miguelcobain/ember-paper/pull/141

### CODE REVIEW

**Comment by brandonjmckay** [comment]
Hm... bad import to the css file.

**Comment by brandonjmckay** [comment]
How do I find the path to the scss files?

**Comment by DanChadwick** [comment]
You're missing entires in `ember-paper/index.js`, `treeForStyles()`. Not sure how this could be working for you locally.

File: addon/components/paper-toast-bounds.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  tagName: 'div',
```
Comment (DanChadwick) [inline addon/components/paper-toast-bounds.js]: div is the default, right?

File: addon/components/paper-toast.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+import Ember from 'ember';
+import TransitionMixin from 'ember-css-transitions/mixins/transition-mixin';
+import PaperToastBounds from './paper-toast-bounds';
+const {
+  Component,
+  computed,
+  run
+} = Ember;
+const {
+  later,
```
Comment (DanChadwick) [inline addon/components/paper-toast.js]: We decided to not destructure past the first level. See contributing.md. (If it isn't up-to-date, let me know). So you would code `run.later(`, not `later`.

File: addon/components/paper-toast.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  },
+
+  addDestroyedElementClone(parent, index, clone) {
+   parent.append(clone);
+  },
+
+  onSwipe(ev) {
+    //Add swipeleft/swiperight class to element so it can animate correctly
+    this.$().addClass(`md-${ev.type}`);
+    this.sendAction('on-close');
```
Comment (DanChadwick) [inline addon/components/paper-toast.js]: See contributing.md re action names. `onClose` is the new `on-close`, for example.

File: addon/components/paper-toast.js (hunk 2 above)
Comment (brandonjmckay) [inline addon/components/paper-toast.js]: Ok, I can change that.

File: app/styles/paper-toast-extras.scss [hunk 4]
```diff
@@ -0,0 +1,6 @@
+// Ember-paper extras for paper-toast
```
Comment (DanChadwick) [inline app/styles/paper-toast-extras.scss]: There should be no styles within `ember-paper`. They should all come from `angular-source-material`. After you fix the import issue, hopefully you can just remove these and the angular styles will match your markup.

File: app/styles/paper-toast.scss [hunk 5]
```diff
@@ -0,0 +1,171 @@
+// See height set globally, depended on by buttons
```
Comment (DanChadwick) [inline app/styles/paper-toast.scss]: Ditto.

File: app/templates/components/paper-toast.hbs [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{{#if hasBlock}}
+  {{yield}}
+
```
Comment (DanChadwick) [inline app/templates/components/paper-toast.hbs]: Remove extra blank lines. I don't think they are need for clarity.

File: app/templates/components/paper-toast.hbs [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+{{else}}
+    <span flex="">{{message}}</span>
```
Comment (DanChadwick) [inline app/templates/components/paper-toast.hbs]: Indentation is too deep.

File: addon/components/paper-toast-bounds.js (hunk 1 above)
Comment (brandonjmckay) [inline addon/components/paper-toast-bounds.js]: Yes, it would be. I missed this file, will update.

File: tests/dummy/app/router.js [hunk 8]
```diff
@@ -29,6 +29,7 @@ Router.map(function() {
   this.route('switch');
   this.route('typography');
   this.route('toolbar');
+  this.route('toast');
```
Comment (DanChadwick) [inline tests/dummy/app/router.js]: These should be alphabetical. Looks like toolbar is wrong too.

File: tests/dummy/app/templates/application.hbs [hunk 9]
```diff
@@ -33,6 +45,7 @@
         {{#paper-item onClick=(transition-to "slider")}}Slider {{paper-icon "warning"}}{{/paper-item}}
         {{#paper-item onClick=(transition-to "switch")}}Switch{{/paper-item}}
         {{#paper-item onClick=(transition-to "toolbar")}}Toolbar {{paper-icon "warning"}}{{/paper-item}}
+        {{#paper-item onClick=(transition-to "toast")}}Toast{{/paper-item}}
```
Comment (DanChadwick) [inline tests/dummy/app/templates/application.hbs]: Alphabetical, please.

File: tests/dummy/app/templates/toast.hbs [hunk 10]
```diff
@@ -0,0 +1,179 @@
+{{#paper-toolbar}}
+    <h2 class="md-toolbar-tools">
+      {{#paper-sidenav-toggle class="menu-sidenav-toggle"}}
+        {{paper-icon icon="menu"}}
+      {{/paper-sidenav-toggle}}
+        <span>Toasts</span>
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: Indentation

File: tests/dummy/app/templates/toast.hbs [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+    {{#paper-toast-bounds}}
+        <h3>Basic Usage</h3>
+        <p>Toast can be dismissed with a swipe, a timer, or a button.</p>
+      {{#paper-button class="md-fab md-fab-top-right"}}FAB{{/paper-button}}
+
+      <!-- some toasts -->
+
+      {{#if isOpenCustom}}
+        {{#paper-toast
+        on-close="closeCustom"
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: Indentation

File: tests/dummy/app/templates/toast.hbs [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+        left=left right=right top=top bottom=bottom as |toast|}}
+            <span flex="">Custom template</span>
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: Indentation

File: tests/dummy/app/templates/toast.hbs [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+          {{#paper-button warn=true action="closeCustom"}}Close{{/paper-button}}
+        {{/paper-toast}}
+      {{/if}}
+
+      {{#if isOpenDelayed}}
+        {{paper-toast hide-delay=4000
+        on-close="closeDelayed"
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: Indentation. The `{{` and closing `}}` should be at the same level.

File: tests/dummy/app/templates/toast.hbs [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+        message="Some message here."}}
+      {{/if}}
+
+      <!-- toasts end -->
+
+
+      <div layout-fill layout="column">
+          <div layout="row" layout-sm="column" layout-align="space-around" style=" padding: 0 50px 0 50px;">
+            {{#paper-button action="openBasic"}}
+                Basic
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: Indentation

File: tests/dummy/app/templates/toast.hbs [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+          </div>
+      </div>
+
+      <h3>Position</h3>
+      {{#paper-checkbox checked=bottom}}Bottom{{/paper-checkbox}}
+      {{#paper-checkbox checked=left}}Left{{/paper-checkbox}}
+      {{#paper-checkbox checked=right}}Right{{/paper-checkbox}}
+      {{#paper-checkbox checked=top}}Top{{/paper-checkbox}}
+
+
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: Remove extra blank lines.

File: tests/dummy/app/templates/toast.hbs [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+    <div class="doc-content">
+
+      <h3>Global toast</h3>
+
+      {{#paper-button raised=true primary=true  action="saveSomething"}}
+          Global toast
+      {{/paper-button}}
+
+      <p>The toast must be inside some div that have enough height to display it. For example the toast can be placed
```
Comment (DanChadwick) [inline tests/dummy/app/templates/toast.hbs]: One long line, or `<p>` and `</p>` alone and at same indentation, please.

**Comment by DanChadwick** [comment]
Also, run jscs as described in the contributing guide, if you haven't already. Travis found some things.

File: app/templates/components/paper-toast.hbs (hunk 6 above)
Comment (brandonjmckay) [inline app/templates/components/paper-toast.hbs]: I personally think they make the file more readable by separating the parts into chunks. If that is required by the style guide I will change it though.

**Comment by brandonjmckay** [comment]
@DanChadwick How is this now?

**Comment by canufeel** [comment]
@DanChadwick That is cool new feature in my opinion! I was lacking it in some projects badly. Any idea what is blocking the merge? I am ready to help and deal with issues.

**Comment by DanChadwick** [comment]
@canufeel This needs review by @miguelcobain, who has a backlog of tasks that relate to fixing existing components. I know a few tiny things that need fixing in this PR, but I hesitate to note them until we know that Miguel will have time to look at it. Such is live in open source land. :-1:

**Comment by jamesdixon** [comment]
@DanChadwick is anyone else authorized (wasn't sure if you were a collaborator at the time of writing) to review this so we can get the conflicts fixed the PR merged?

**Comment by DanChadwick** [comment]
James, Miguel wants to finish the existing 1.0 components first. My role is mostly admin helper.

**Comment by cs3b** [comment]
It looks like this ticket should be closed as it is replaced by #529 ?

**Comment by brandonjmckay** [comment]
@canufeel check out [ember-notify](https://github.com/aexmachina/ember-notify) good way to go.
When it gets built out for ember-paper for realz I would be surprised if they don't leverage it.

**Comment by v3ss0n** [comment]
@brandonjmckay  i do not see ember-notify's interest in ember-paper at all.

**Comment by miguelcobain** [comment]
see https://github.com/miguelcobain/ember-paper/pull/529#issuecomment-280641296 for the direction ember-paper will take

**Pull request closed without merging**